invenio_deposit/api.py

```diff
@@ -131,12 +133,15 @@ def fetch_published(self):
         return resolver.resolve(pid_value)
 
     def merge_with_published(self):
-        """."""
-        # TODO if revisions are the same then we can apply directly
-        # latest = self.fetch_published()
-        # initial = A.revisions[self['_deposit']['pid']['revision']]
-        # patch = diff(initial, self)
-        # latest.apply(patch)
+        """Merge changes with latest published version."""
+        latest_id, latest_published = self.fetch_published()
+        current = dict(self.dumps())
+        before = dict(self.revisions[-2].dumps())
```

Comment:
ah, ok.. now I understand. The `published` was from you first comment..
I didn't understand because in the code you wrote the `published` variable was renamed in `first` when defined but one line after you refer to `published`.